### PR TITLE
feat(package): remove the need for separate vm and browser imports, upgrade deps

### PR DESCRIPTION
**Pull request by michaelcarter-wf**
## Ultimate problem:
- Consumers need to import different files and run a configuration function to use json_schema on vm or browser.

## How it was fixed:
- Use conditional imports to do it under the hood with no config.

## Notes:
- Configuration functions still exist but are deprecated no-ops
- Global assessors still exist, but target the conditional imports.

## Potential areas of regression:

---

> __FYA:__ @michaelcarter-wf

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/json_schema/json_schema.dart [hunk 1]
```diff
@@ -194,13 +197,10 @@ class JsonSchema {
...
   static Future<JsonSchema> createSchemaFromUrl(String schemaUrl, {SchemaVersion schemaVersion}) {
-    if (globalCreateJsonSchemaFromUrl == null) {
-      throw StateError('no globalCreateJsonSchemaFromUrl defined!');
-    }
-    return globalCreateJsonSchemaFromUrl(schemaUrl, schemaVersion: schemaVersion);
+    return createClient()?.createSchemaFromUrl(schemaUrl, schemaVersion: schemaVersion);
   }
 
-  /// Construct and validate a JsonSchema.
+  /// Construct and validate a Jsonchema.
```
Comment (bryanrezende-wf) [inline lib/src/json_schema/json_schema.dart:203]: Small typo

File: lib/src/json_schema/schema_url_client/html_schema_url_client.dart [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+class HtmlSchemaUrlClient extends SchemaUrlClient {
+  @override
+  createSchemaFromUrl(String schemaUrl, {SchemaVersion schemaVersion}) async {
+    final uriWithFrag = Uri.parse(schemaUrl);
+    var uri = uriWithFrag.removeFragment();
+    if (schemaUrl.endsWith('#')) {
+      uri = uriWithFrag;
+    }
+    if (uri.scheme != 'file') {
+      // _logger.info('Getting url $uri'); TODO: re-add logger.
```
Comment (bryanrezende-wf) [inline lib/src/json_schema/schema_url_client/html_schema_url_client.dart:19]: Do we want this done before merging?

**Review by bryanrezende-wf — CHANGES_REQUESTED** [review]

File: lib/src/json_schema/schema_url_client/html_schema_url_client.dart (hunk 2 above)
Comment (michaelcarter-wf) [reply lib/src/json_schema/schema_url_client/html_schema_url_client.dart:19]: No

**Review by bryanrezende-wf — APPROVED** [review]
+1

**Review by bryanrezende-wf — APPROVED** [review]
+1

**Comment by michaelcarter-wf** [comment]
+10 

- [x] CI Passes
- [x] Tests added for all deprecated functions

**Review by bryanrezende-wf — APPROVED** [review]
+1 to latest test addition

**Review by jordanross-wf — APPROVED** [review]
+1 @Workiva/release-management-pp

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**